### PR TITLE
read-only flag in EncryptedFs introduced

### PR DESCRIPTION
**Pull request by someotherself**
Implementation for https://github.com/radumarias/rencfs/issues/168

### CODE REVIEW

File: src/encryptedfs.rs [hunk 1]
```diff
@@ -570,6 +572,7 @@ pub struct EncryptedFs {
     sizes_write: Mutex<HashMap<u64, AtomicU64>>,
     sizes_read: Mutex<HashMap<u64, AtomicU64>>,
     requested_read: Mutex<HashMap<u64, AtomicU64>>,
+    read_only: Mutex<bool>,
```
Comment (radumarias) [inline src/encryptedfs.rs]: maybe we don't need mutex as we will not change it

File: src/encryptedfs.rs [hunk 2]
```diff
@@ -634,6 +638,10 @@ impl EncryptedFs {
         Ok(arc)
     }
 
+    pub async fn read_only(&self, read_only: bool) {
```
Comment (radumarias) [inline src/encryptedfs.rs]: don't need this, it's enough to set it one time in `new`

File: src/encryptedfs.rs [hunk 3]
```diff
@@ -646,6 +654,11 @@ impl EncryptedFs {
         self.contents_path(ino).is_file()
     }
 
+    async fn is_read_only(&self) -> bool {
```
Comment (radumarias) [inline src/encryptedfs.rs:654]: this is fine to keep

File: src/encryptedfs.rs [hunk 4]
```diff
@@ -667,6 +680,12 @@ impl EncryptedFs {
         if self.exists_by_name(parent, name)? {
             return Err(FsError::AlreadyExists);
         }
+        // let read_only_guard = self.read_only.lock().await;
+        // if *read_only_guard {
+        // }
+        if self.is_read_only().await {
```
Comment (radumarias) [inline src/encryptedfs.rs:679]: this is a read only operation, should work

**Review by radumarias — CHANGES_REQUESTED** [review]
Added which methods should ret `Err`, all other should work https://github.com/radumarias/rencfs/issues/168

File: src/encryptedfs.rs (hunk 1 above)
Comment (someotherself) [reply src/encryptedfs.rs]: Alright. read_only is added an an arguement now.

File: src/encryptedfs.rs (hunk 2 above)
Comment (someotherself) [reply src/encryptedfs.rs]: This is deleted now.

File: src/encryptedfs.rs (hunk 4 above)
Comment (someotherself) [reply src/encryptedfs.rs:679]: Removed as well.

**Review by radumarias — APPROVED** [review]
I you added to all the places from the issue then it's fine

**Review by radumarias — APPROVED** [review]

**Pull request merged**